azkaban-common/src/main/java/azkaban/project/FlowLoaderFactory.java

```diff
@@ -0,0 +1,59 @@
+/*
+* Copyright 2017 LinkedIn Corp.
+*
+* Licensed under the Apache License, Version 2.0 (the “License”); you may not
+* use this file except in compliance with the License. You may obtain a copy of
+* the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an “AS IS” BASIS, WITHOUT
+* WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+* License for the specific language governing permissions and limitations under
+* the License.
+*/
+
+package azkaban.project;
+
+import static java.util.Objects.requireNonNull;
+
+import azkaban.Constants;
+import azkaban.utils.Props;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/**
+ * Factory class to generate flow loaders.
+ */
+@Singleton
```

Comment:
Does it need to be a singleton?

Reply:
We only need one FlowLoaderFactory instance, so I think it can be singleton. Any reason not to use singleton here?

Reply:
See https://github.com/google/guice/wiki/Scopes
"Singletons are popular in Java applications but they don't provide much value, especially when dependency injection is involved. Although singletons save object creation (and later garbage collection), initialization of the singleton requires synchronization; getting a handle to the single initialized instance only requires reading a volatile. Singletons are most useful for:

stateful objects, such as configuration or counters
objects that are expensive to construct or lookup
objects that tie up resources, such as a database connection pool."